desc/objectives/_geometry.py

```diff
@@ -242,7 +245,7 @@ def build(self, use_jit=True, verbose=1):
         else:
             grid = self._grid
 
-        self._dim_f = 1
+        self._dim_f = grid.num_zeta
```

Comment:
this is not the correct dimension, if the default grid returns the max across all `zeta`.